### DISCUSSION TITLE
Problems getting info from specific user

### DISCUSSION THREAD

**Discussion by Bysev** [trakt/trakt-api#344] [Q&A] [answered]
Hi!
   As most questions starts here, please bear with me as I'm learning how to work with API (and Python).

I am working on a python module for my friends and I and am having lots of trouble accessing information about one of my friend's account.

Me:
https://trakt.tv/users/bysev

My Friend:
https://trakt.tv/users/clouds-26cf6484-4fc2-4c9e-968e-36ff894a268f

When I try to call for any of his information I get an empty response. It looks like I get a 200 so it should be ok but there is nothing in it. 

I tried with or without oauth, I checked what I could see from his profile when I am logged out of the website and then to call it from my module and I see nothing.

The problem is, I know my oauth works correctly because I get to see things in my personnal profile that are set to private, I can see my other friends' lists that are only shared with friends, but this one, nothing.

He is the only one though that has such a weird web address. I took for granted that his slug is 'clouds-26cf6484-4fc2-4c9e-968e-36ff894a268f' is this where I am wrong?

TL;DR: I guess my question is: can you guys access ANY information about the user reached with this web address?

https://trakt.tv/users/clouds-26cf6484-4fc2-4c9e-968e-36ff894a268f

Thanks a lot!

**Comment by tysonkerridge** [comment, marked as answer]
Using that slug appears to work for me, with the following request/headers:
```
GET https://api.trakt.tv/users/clouds-26cf6484-4fc2-4c9e-968e-36ff894a268f?extended=full
```
```
Content-Type : application/json
trakt-api-version : 2
trakt-api-key : {{your key}}
```

And response:
```
{
    "username": "CloudS.",
    "private": false,
    "name": null,
    "vip": false,
    "vip_ep": false,
    "ids": {
        "slug": "clouds-26cf6484-4fc2-4c9e-968e-36ff894a268f"
    },
    "joined_at": "2022-04-17T22:16:20.000Z",
    "location": null,
    "about": null,
    "gender": null,
    "age": null,
    "images": {
        "avatar": {
            "full": "https://secure.gravatar.com/avatar/2499ed5730594de6045dc36a971fa957?d=https%3A%2F%2Fwalter.trakt.tv%2Fhotlink-ok%2Fplaceholders%2Fmedium%2Fzoidberg.png&r=pg&s=256"
        }
    }
}
```

**Reply by Bysev** [reply]
Thanks for your fast answer!
I do get this information too, but let's say I want to see their watchlist. Even logged out on the website we can see they have 'Matrix: Resurrected' in their watchlist, when I try to call their watchlist for movies through the API I get nothing.

**Reply by tysonkerridge** [reply]
Perhaps you're using the wrong endpoint? I'm assuming you're using the API docs? https://trakt.docs.apiary.io/
This appears to work:
```
GET https://api.trakt.tv/users/clouds-26cf6484-4fc2-4c9e-968e-36ff894a268f/watchlist
```
```
Content-Type : application/json
trakt-api-version : 2
trakt-api-key : {{your key}}
```
Response:
```
[
    {
        "rank": 1,
        "id": 687910322,
        "listed_at": "2022-04-18T14:16:57.000Z",
        "notes": null,
        "type": "movie",
        "movie": {
            "title": "The Matrix Resurrections",
            "year": 2021,
            "ids": {
                "trakt": 466465,
                "slug": "the-matrix-resurrections-2021",
                "imdb": "tt10838180",
                "tmdb": 624860
            }
        }
    }
]
```

**Reply by Bysev** [reply]
Oh thanks for showing me it works, I'll get into it a little more. I've been using PyTrakt to help me with it but it looks like there must be something different behind the scenes. 

I'll go check how they use it and try to modify it if I find what doesn't work.

Thanks again

**Reply by tysonkerridge** [reply]
Ah okay, yeah I'm not 100% familiar with Python but it looks like it might be using the username instead of the slug based on line 333 of https://github.com/moogar0880/PyTrakt/blob/22a1066601b1e30947a8b11eaa911f2b6ef5b04e/trakt/users.py
```python
data = yield 'users/{username}/watchlist/movies'.format(
    username=slugify(self.username),
)
```
Perhaps if you can get it to take the slug of the user's ids rather than the username that should work.

**Reply by Bysev** [reply]
I went into debug mode and slowing down the process to watch the steps got me the right result.

I thought I had it solved but as soon as I ran it full speed the problem showed up again...

**Reply by Bysev** [reply]
So I have tried with what was provided directly on the trakt api website, it worked! 
The only thing that was making me nervous with this is the whole oauth token that I still got from PyTrakt.

For now the token I have works so I guess it's a problem for me from the future!

thanks again.

**Reply by tysonkerridge** [reply]
Assuming you used your own API key/secret etc and then authed with your account you should be able to revoke access in your account settings anyway https://trakt.tv/oauth/authorized_applications. Good luck 😀
